tests/annotations/tests.py

```diff
@@ -550,21 +550,6 @@ def test_values_with_pk_annotation(self):
         for publisher in publishers.filter(pk=self.p1.pk):
             self.assertEqual(publisher["book__rating"], publisher["total"])
 
-    @skipUnlessDBFeature("allows_group_by_pk")
-    def test_rawsql_group_by_collapse(self):
-        raw = RawSQL("SELECT MIN(id) FROM annotations_book", [])
-        qs = (
-            Author.objects.values("id")
-            .annotate(
-                min_book_id=raw,
-                count_friends=Count("friends"),
-            )
-            .order_by()
-        )
-        _, _, group_by = qs.query.get_compiler(using="default").pre_sql_setup()
-        self.assertEqual(len(group_by), 1)
-        self.assertNotEqual(raw, group_by[0])
-
```

Comment:
This can now easily be achieved by using a `Subquery` which will do the right thing.